### PR TITLE
rework modifier key, Code Cleanup Sandbag, Code Cleanup Tripod

### PR DESCRIPTION
**Pull request by commy2**
https://github.com/acemod/ACE3/issues/1675

Currently the modifier key doesn't work at all with the default key bind.

| | ctrl not pressed | ctrl pressed
------------- | ------------- | -------------
| dragging | - | -
| carrying | lowering/raising | -
| explosives | - | rotate
| sandbag | - | rotate
| tripod | lowering/raising | -
| tactical ladder | extend | tilt

- [x] dragging
- [x] carrying
- [x] explosives
- [x] sandbag
    - [x] some missing event handling: dc, respawn, killed, unconscious, captive, surrendered
    - [x] check if sandbag still in inventory during placement
- [x] tripod
    - [x] some missing event handling: dc, respawn, killed, unconscious, captive, surrendered
    - [x] correct string
    - [x] distance check while adjusting
    - [x] move placement option out of "Equipment"
- [x] tactical ladder
    - [x] some missing event handling: dc, respawn, killed, unconscious, captive, surrendered
    - [x] correct string

- [ ] doc on class name: http://ace3mod.com/wiki/missionmaker/class-names.html#tactical-ladder
- [ ] MP testing with hard dc, unconsciousness, stealing the bag from the backpack etc.

### CODE REVIEW

**Comment by PabstMirror** [comment]
This seems like a good solution, should also fix #1235

CBA had an issue for this, but it's closed https://github.com/CBATeam/CBA_A3/issues/77

btw `displayAddEventHandler` will take CODE or STRING.

**Comment by commy2** [comment]
>btw displayAddEventHandler will take CODE or STRING.

Yup, it does for "keyDown" and "keyUp", but it doesn't for "mouseZChanged" which I copied it from. Fixed now.

File: addons/common/functions/fnc_handleModifierKeyInit.sqf [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+ * None
+ *
+ * Return Value:
+ * None
+ *
+ * Public : No
+ */
+#include "script_component.hpp"
+
+(findDisplay 46) displayAddEventHandler ["KeyDown", {_this call FUNC(handleModifierKey)}];
```
Comment (jokoho48) [inline addons/common/functions/fnc_handleModifierKeyInit.sqf]: why not using `FUNC(handleModifierKey)`

**Comment by commy2** [comment]
- [ ] some rewriting in EFUNC(interaction,showMouseHint);

**Comment by commy2** [comment]
I'm using CorruptedHeart's wonderfull mouse hint's to make it more clear for user what they can do with the stuff they are interacting with.



Has anyone a nice idea on how to improve this design to make it as user friendly as possilbe? Besides making the control bigger, so the text is not cut off of course.

File: addons/common/functions/fnc_handleModifierKeyInit.sqf (hunk 1 above)
Comment (commy2) [inline addons/common/functions/fnc_handleModifierKeyInit.sqf]: done

**Comment by SilentSpike** [comment]
No ideas currently about the UI itself, but for carrying I notice you've listed `ctrl` to `raise/lower`. Isn't that inconsistent with elsewhere?

**Comment by MikeMatrix** [comment]
I think an improvement could be to make it more destinct on what is keybind and what is the action it takes. A bit of spacing and the order of things (Modifier keys in front) can go a long way of making it look nicer and more understandable.

**Comment by commy2** [comment]
- [ ] fix being able to throw grenades when carrying / dragging.

**Comment by commy2** [comment]
Done. Needs some MP testing with getting killed/kocked out and switching units and stuff.

**Pull request closed without merging**